specification/vmware/Microsoft.AVS/client.tsp

```diff
@@ -0,0 +1,80 @@
+);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(Azure.ResourceManager.ProxyResource.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(Azure.ResourceManager.Foundations.ProxyResourceUpdateModel.properties
+);
+
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(PrivateCloudUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(ClusterUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(WorkloadNetworkSegmentUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(WorkloadNetworkDhcpUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(WorkloadNetworkPortMirroringUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(WorkloadNetworkVMGroupUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(WorkloadNetworkDnsServiceUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(WorkloadNetworkDnsZoneUpdate.properties);
+#suppress "deprecated" "property flatten for SDK backward compatibility"
+@@flattenProperty(PlacementPolicyUpdate.properties);
+
+// operation
+@@friendlyName(Addons.list, "list");
+@@friendlyName(PlacementPolicies.list, "list");
+@@friendlyName(VirtualMachines.list, "list");
+@@friendlyName(HcxEnterpriseSites.list, "list");
+@@friendlyName(ScriptExecutions.list, "list");
+@@friendlyName(ScriptPackages.list, "list");
+@@friendlyName(Datastores.list, "list");
+@@friendlyName(Clusters.list, "list");
+@@friendlyName(ScriptCmdlets.list, "list");
+@@friendlyName(Authorizations.list, "list");
+@@friendlyName(CloudLinks.list, "list");
+@@friendlyName(GlobalReachConnections.list, "list");
+@@friendlyName(WorkloadNetworks.list, "list");
+
+// model
+@@clientName(Azure.ResourceManager.Foundations.SystemAssignedServiceIdentity,
+  "PrivateCloudIdentity"
+);
+@@clientName(Azure.ResourceManager.Foundations.SystemAssignedServiceIdentityType,
+  "ResourceIdentityType"
+);
+
+@@clientName(VmVmPlacementPolicyProperties, "VmPlacementPolicyProperties");
+@@clientName(WorkloadNetworkPublicIP, "WorkloadNetworkPublicIp");
+@@clientName(WorkloadNetworkPublicIPProvisioningState,
+  "WorkloadNetworkPublicIpProvisioningState"
+);
+
+// property
+@@clientName(Circuit.expressRouteID, "expressRouteId");
+@@clientName(Circuit.expressRoutePrivatePeeringID,
+  "expressRoutePrivatePeeringId"
+);
+@@clientName(WorkloadNetworkPublicIPProperties,
+  "WorkloadNetworkPublicIpProperties"
+);
+@@clientName(WorkloadNetworkPublicIPProperties.publicIPBlock, "publicIpBlock");
```

Comment:
These to make it compatible m4 prenamer (different SDK may have different configure).